### PR TITLE
Add events for task.created and task.updated

### PR DESCRIPTION
**Pull request by jsun-m**
I'm looking for opinions on where to push events for these backend transactions. 
Rather than updating in 15 places where `backendRepo.UpdateTask` is called, I currently have the event happen after the database transaction is fulfilled.

Alternative to this, I can write a layer on top of `backendRepo` such as `backendRepoWithEvents` or something similar that wraps functions such as 
- `UpdateTask`
- `CreateTask`
- and other places where we might want events following a database transaction such as `CreateDeployment`

### CODE REVIEW

File: pkg/types/event.go [hunk 1]
```diff
@@ -15,6 +19,25 @@ var (
...
+		stub.deployed
+		stub.served
+
+		container.lifecycle.updated
+		worker.lifecycle.updated
+
+		the container.metrics event is slightly different because they are snapshots of a resource not a change in state
+		we might need to altered this or rethink concept of an event
+	*/
+	EventTaskUpdated = "task.update"
```
Comment (luke-lombardi) [inline pkg/types/event.go:39]: we should make this consisent, if it's past tense / present tense

File: pkg/repository/base.go [hunk 2]
```diff
@@ -149,6 +149,8 @@ type EventRepository interface {
 	PushDeployStubEvent(workspaceId string, stub *types.Stub)
 	PushServeStubEvent(workspaceId string, stub *types.Stub)
 	PushRunStubEvent(workspaceId string, stub *types.Stub)
+	PushTaskUpdatedEvent(task *types.TaskWithRelated)
```
Comment (luke-lombardi) [inline pkg/repository/base.go:152]: 👍

File: Makefile [hunk 3]
```diff
@@ -42,7 +42,7 @@ proxy:
 
 runner:
 	for target in py312 py311 py310 py39 py38; do \
-		docker build . --no-cache --target $$target --platform=linux/amd64 -f ./docker/Dockerfile.runner -t localhost:5001/beta9-runner:$$target-$(runnerTag); \
```
Comment (luke-lombardi) [inline Makefile:45]: put back

**Review by luke-lombardi — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (jsun-m) [reply Makefile:45]: I think i added this a while back when i was compiling workers for lambda locally. It dramatically increases compile time when building workers locally but was necessary before adding the github action

Comment (luke-lombardi) [reply Makefile:45]: it's necessary for building the correct arch on m1s / arm hardware

Comment (luke-lombardi) [reply Makefile:45]: this is for the runner image, not for the worker

**Review by luke-lombardi — CHANGES_REQUESTED** [review]
left one more comment then good to go

**Review by luke-lombardi — APPROVED** [review]

**Pull request merged**